### PR TITLE
[Cleanup] Remove unused `normalize_reward` codepath in CriticModel

### PR DESCRIPTION
**Pull request by SumanthRH**
# What does this PR do?

We have this unused `normalize_reward` codepath in CriticModel. This PR removes it and also improves some typing. 

The original code here is from OpenRLHF. The code has been a no-op at the moment - this is because the mean and std buffers are intiialized to zero and 1, and used as-is during training.

### CODE REVIEW

File: skyrl-train/skyrl_train/workers/worker.py [hunk 1]
```diff
@@ -328,16 +331,19 @@ def forward(
         returns: torch.Tensor,
         loss_mask: Optional[torch.Tensor] = None,
     ) -> torch.Tensor:
+
         if self.clip_eps is not None:
             values_clipped = old_values + (values - old_values).clamp(-self.clip_eps, self.clip_eps)
             surr1 = (values_clipped - returns) ** 2
             surr2 = (values - returns) ** 2
             loss = torch.max(surr1, surr2)
+            clipfrac = masked_mean((surr1 > surr2).float(), loss_mask).mean().detach().item()
```
Comment (SumanthRH) [inline skyrl-train/skyrl_train/workers/worker.py:340]: tiny improvement by calculating values clipfrac as well for logging

File: skyrl-train/skyrl_train/distributed/strategy.py [hunk 2]
```diff
@@ -14,11 +14,11 @@ class DistributedStrategy(ABC):
     def setup_distributed(self):
         pass
 
-    def all_reduce(self, data, op="mean"):
+    def all_reduce(self, data, op="mean") -> Dict[str, Any]:
```
Comment (tyler-griggs) [inline skyrl-train/skyrl_train/distributed/strategy.py]: In our strategy code for fsdp and deepspeed, this (and `all_gather`) can return either a Dict or a Tensor

**Review by tyler-griggs — APPROVED** [review]

File: skyrl-train/skyrl_train/distributed/strategy.py (hunk 2 above)
Comment (SumanthRH) [reply skyrl-train/skyrl_train/distributed/strategy.py]: Fixed!
